src/endless/Eosldr.cpp

```diff
@@ -0,0 +1,204 @@
+#include "stdafx.h"
+#include "Eosldr.h"
+#include "GeneralCode.h"
+#include "WMI.h"
+
+// Paths for g2ldr binaries, used to chainload GRUB from the Windows Boot
+// Manager on BIOS systems, rather than the other way round.
+#define EOSLDR_SUBDIRECTORY             L"eosldr"
+// Stub loaded by WBM
+#define EOSLDR_MBR_FILE                 L"eosldr.mbr"
+// GRUB core.img with a special preamble, loaded by eosldr.mbr
+#define EOSLDR_IMG_FILE                 L"eosldr"
+
+// GUID for Endless OS's entry in the BCD boot menu, stored in the Uninstall
+// registry key
+#define REGKEY_BCD_GUID         L"EndlessBcdGuid"
```

Comment:
I'm not sure what will happen if you have two Windows installations. Would you expect to be able to install two separate copies of Endless OS? (It wouldn't work if both Windowses are on the same physical disk, but it might work if you have two separate disks because our GRUB scopes its search to `$bootdev`.) If so, would you expect our installer to add both to both boot menus? To be honest, I don't know how the Windows boot menu works on such systems. But that's the scenario I had in mind when creating a fresh GUID each time.

OTOH, we could always use a well-known GUID for now, and if we do end up wanting to support >1 Endless OS menu entry, reintroduce this idea of generating a fresh GUID and storing it in the registry; on uninstall, try to get the GUID from the registry, and fall back to the well-known one.